ledger-api/grpc-definitions/com/digitalasset/admin/api/v1/party_management_service.proto

```diff
@@ -0,0 +1,99 @@
+
+option java_outer_classname = "PartyManagementServiceOuterClass";
+option java_package = "com.digitalasset.admin.api.v1";
+
+// Allows clients to query the DAML-LF packages that are supported by the server.
+service PartyManagementService {
+
+  // Returns the identifiers of all parties hosted by the server.
+  rpc ListParties (ListPartiesRequest) returns (ListPartiesResponse);
+
+  // Adds a party to the set hosted by the server. The validations performed
+  // are implementation specific. For instance, party names might be checked
+  // if they follow a certain pattern or the operation may be vetted by the 
+  // consensus of the collaborating nodes, etc.
+  rpc AddParty (AddPartyReuest) returns (AddPartyResponse);
+}
+
+message ListPartiesRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Required
+  string ledger_id = 1;
+
+  // TODO: Discuss teh necessity to include below fields, they are
+  // present in the canton version of this service.
+  
+  // Timestamp seems irrelevant. Offset would be more interesting, but
+  // it could incentivise concurrent use of this api and ledger api,
+  // which should be rather discouraged.
+  //google.protobuf.Timestamp asOf = 1;
+
+  // Domain is implicit in most implementations, ledger_id comes
+  // close to this field semantically
+  //string filterDomain = 2;
+
+  // Party filter could actually be interesting one to include.
+  // I'd prefer to skip it for KISS principle's sake
+  //string filterParty = 3;
+}
+
+message ListPartiesResponse {
+
+  // The details of all DAML parties hosted by the server.
+  // Required
+  repeated PartyDetails party_details = 1;
+}
+
+message PartyDetails {
+
+  // The IDs of a DAML party hosted by the server.
+  // Required
+  string party_ids = 1;
+
+  // TODO: Discuss the necessity to include below fields, they are
+  // present in the canton version of this service.
+  // All those fields could be added later if need be.
+  
+  // Domain is implicit in most of other implementations. There is
+  // only one. 
+  //string domain = 2;
+
+  // Participant is implicit in most of other implementations. It is the
+  // one you are communicating with.
+  //string participant = 3;
+
+  // Privilege is implicit in most of other implementations. It is a
+  // submitting participant.
+  //com.digitalasset.canton.identity.v0.ParticipantPrivilege privilege = 4;
+
+  // TrustLevel is implicit in most of other implementations. It is a 
+  // regular participant.
+  //com.digitalasset.canton.identity.v0.TrustLevel trustLevel = 5;
+}
+
+message AddPartyRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Required
+  string ledger_id = 1;
+
+  // Party to be added to the ledger api server
+  // Required
+  string party_id = 2;
+
+  // The id of the agent making the change. Interpretation is implementation specific.
+  // Optional
+  string added_by = 3;
```

Comment:
What's an agent?

This API doesn't really make any sense to me as it is now. Shouldn't there be some implementation specific metadata here? E.g. adding a party might also mean that you provide your public key etc.

Reply:
The problem with public key is that in many cases it is implementation specific. Canton exposes the keys through their version of this api. Sawtooth on-boards participants natively. You still provide your key on the consensus layer when you want to store the party to participant data from this call in the KVs. The key, however, does not leak out to the client-facing apis. The more I think about it, the more I am convinced we need a participant identity. Question is how to implement it, so that it is a) stable until the end of time b) is available to a participant before it connects to the consensus layer (from which it must filter own party mapping)

Reply:
Yep, adding parties in Canton looks like this:
```
da authorize_party_to_participant(Add, "identityKey", RequestSide.Both, "Marcin::<fingerprint1>", "participant2::<fingerprint2>", ParticipantPrivilege.Observation)
```
so my issue is that i can't squeeze things like `ParticipantPrivilege` and `RequestSide` into this call. 
What I could imagine is that this `AddParty` becomes a "suggestion" and then the participant node internally can decide on how to deal with such requests, but you can't bake too many assumptions on how such requests are processed into `daml-on-x`